### PR TITLE
Feature/pok 63 sheet row parser unit tests

### PR DESCRIPTION
**Pull request by Finrod-Amandil**

### CODE REVIEW

File: PokeOneWeb.DataSync.Tests/GoogleSpreadsheet/Import/Impl/Sheets/ElementalTypes/ElementalTypeSheetRowParserTest.cs [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+            // Act
+            var actual = () => parser.ReadRow(values);
+
+            // Assert
+            actual.Should().Throw<InvalidRowDataException>();
+        }
+
+        [Theory]
+        [InlineData("")] // Name must be non-empty
```
Comment (frei-sven) [inline PokeOneWeb.DataSync.Tests/GoogleSpreadsheet/Import/Impl/Sheets/ElementalTypes/ElementalTypeSheetRowParserTest.cs]: Could also be a Fact if only one inline data is to be tested. This is also in other test files like this.

File: PokeOneWeb.DataSync.Tests/GoogleSpreadsheet/Import/Impl/Sheets/MoveLearnMethodLocations/MoveLearnMethodLocationSheetRowParserTest.cs [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+            result.Should().BeEquivalentTo(expected);
+        }
+
+        [Fact]
+        public void ReadRow_WithAllValidValues_ShouldParse()
+        {
+            // Arrange
+            var parser = new MoveLearnMethodLocationSheetRowParser();
+
+            var moveLearnMethodName = "Move Learn Method Name";
```
Comment (frei-sven) [inline PokeOneWeb.DataSync.Tests/GoogleSpreadsheet/Import/Impl/Sheets/MoveLearnMethodLocations/MoveLearnMethodLocationSheetRowParserTest.cs:49]: Maybe a look at AutoFixture could be something to decrease the amount of time to invest into setup the test data. But I am also quite lazy ;).

File: PokeOneWeb.DataSync.Tests/GoogleSpreadsheet/Import/Impl/Sheets/Pokemon/PokemonSheetRowParserTest.cs [hunk 3]
```diff
@@ -0,0 +1,444 @@
...
+        [InlineData(0, 0, "0", "0", "0", "0", "0", "0", "0", false, "0", 0, "")] // SpecialAttack must be int
+        [InlineData(0, 0, "0", "0", "0", "0", "0", "0", "0", false, "0", 0, 0, "")] // Defense must be int
+        [InlineData(0, 0, "0", "0", "0", "0", "0", "0", "0", false, "0", 0, 0, 0, "")] // SpecialDefense must be int
+        [InlineData(0, 0, "0", "0", "0", "0", "0", "0", "0", false, "0", 0, 0, 0, 0, "")] // Speed must be int
+        [InlineData(0, 0, "0", "0", "0", "0", "0", "0", "0", false, "0", 0, 0, 0, 0, 0, "")] // HitPoints must be int
+        [InlineData(0, 0, "0", "0", "0", "0", "0", "0", "0", false, "0", 0, 0, 0, 0, 0, 0, "")] // AttackEvYield must be int
+        [InlineData(0, 0, "0", "0", "0", "0", "0", "0", "0", false, "0", 0, 0, 0, 0, 0, 0, 0, "")] // SpecialAttackEvYield must be int
+        [InlineData(0, 0, "0", "0", "0", "0", "0", "0", "0", false, "0", 0, 0, 0, 0, 0, 0, 0, 0, "")] // DefenseEvYield must be int
+        [InlineData(0, 0, "0", "0", "0", "0", "0", "0", "0", false, "0", 0, 0, 0, 0, 0, 0, 0, 0, 0, "")] // SpecialDefenseEvYield must be int
```
Comment (frei-sven) [inline PokeOneWeb.DataSync.Tests/GoogleSpreadsheet/Import/Impl/Sheets/Pokemon/PokemonSheetRowParserTest.cs]: Damn son :0

**Review by frei-sven — APPROVED** [review]
So far it looks good. Just some points that can be looked at, which are not necessary to be changed from my pov.

File: PokeOneWeb.DataSync.Tests/GoogleSpreadsheet/Import/Impl/Sheets/ElementalTypes/ElementalTypeSheetRowParserTest.cs (hunk 1 above)
Comment (Finrod-Amandil) [reply PokeOneWeb.DataSync.Tests/GoogleSpreadsheet/Import/Impl/Sheets/ElementalTypes/ElementalTypeSheetRowParserTest.cs]: True, will adjust that.

File: PokeOneWeb.DataSync.Tests/GoogleSpreadsheet/Import/Impl/Sheets/MoveLearnMethodLocations/MoveLearnMethodLocationSheetRowParserTest.cs (hunk 2 above)
Comment (Finrod-Amandil) [reply PokeOneWeb.DataSync.Tests/GoogleSpreadsheet/Import/Impl/Sheets/MoveLearnMethodLocations/MoveLearnMethodLocationSheetRowParserTest.cs:49]: Yes quite a lot of (also sometimes duplicated) lines of code are used to set up test data and I have been thinking about changing something about it mutliple time. But I prefer more code over complicated and obfuscating solutions. With the current solution it is always abundantly clear what the exact test setup is, and it is always possible to change the setup without influencing other tests. But may be a point to discuss.

File: PokeOneWeb.DataSync.Tests/GoogleSpreadsheet/Import/Impl/Sheets/Pokemon/PokemonSheetRowParserTest.cs (hunk 3 above)
Comment (Finrod-Amandil) [reply PokeOneWeb.DataSync.Tests/GoogleSpreadsheet/Import/Impl/Sheets/Pokemon/PokemonSheetRowParserTest.cs]: Yeah its ugly but follows the same principle as all other tests of this kind. Not entirely happy with it either, but havent come up with another solution that I am satisfied with.

**Pull request merged**